tests/impl.h

```diff
@@ -1,5 +1,273 @@
 #ifndef SSE2NEONTEST_H
 #define SSE2NEONTEST_H
+#include "common.h"
+#define ENUM(c) it_##c,
+#define STR(c) #c,
+#define CASE(c)                   \
+    case it_##c:                  \
+        ret = test_##c(*this, i); \
+        break;
+#define TYPE_FOREACH(TYPE)            \
```

Comment:
Can we classify the following instructions as @marktwtn did for SSE, SSE2, SSE3, etc.?